packages/fx-core/src/plugins/solution/fx-solution/solution.ts

```diff
@@ -1062,12 +1078,35 @@ export class TeamsAppSolution implements Solution {
     if (maybePermission.isErr()) {
       return maybePermission;
     }
+
+    const maybeSelectedPlugins = this.getSelectedPlugins(ctx);
+
+    if (maybeSelectedPlugins.isErr()) {
+      return maybeSelectedPlugins;
+    }
+
+    const selectedPlugins = maybeSelectedPlugins.value;
+    const hasFrontend = selectedPlugins?.some((plugin) => plugin.name === PluginNames.FE);
+    const hasBackend = selectedPlugins?.some((plugin) => plugin.name === PluginNames.FUNC);
+    const hasBot = selectedPlugins?.some((plugin) => plugin.name === PluginNames.BOT);
+
+    // load localSettings into context before local debug.
+    const localSettingsProvider = new LocalSettingsProvider(ctx.root);
+    if (await fs.pathExists(localSettingsProvider.localSettingsFilePath)) {
+      ctx.localSettings = await localSettingsProvider.load();
+    } else {
+      ctx.localSettings = await localSettingsProvider.init(hasFrontend, hasBackend, hasBot);
```

Comment:
no need `await` here, since `localSettingsProvider.init` is a sync method?

Reply:
Good catch. Will remove the redundant `await`.